lib/otel/span-processor.js

```diff
@@ -71,12 +73,56 @@ module.exports = class NrSpanProcessor {
       this.reconcileServerAttributes({ segment, span, transaction })
     } else if (span.kind === SpanKind.CLIENT && span.attributes[ATTR_DB_SYSTEM]) {
       this.reconcileDbAttributes({ segment, span })
+    } else if (span.kind === SpanKind.CONSUMER) {
+      this.reconcileConsumerAttributes({ segment, span, transaction })
     } else if (span.kind === SpanKind.PRODUCER) {
       this.reconcileProducerAttributes({ segment, span })
     }
     // TODO: add http external checks
   }
 
+  /**
+   * Detect messaging consumer attributes in the OTEL span and add them
+   * to the New Relic transaction. Note: this method ends the current
+   * transaction.
+   *
+   * @param {object} params
+   * @param {object} params.span The OTEL span entity that possibly contains
+   * desired attributes.
+   * @param {Transaction} params.transaction The NR transaction to attach
+   * the found attributes to.
+   */
+  reconcileConsumerAttributes({ span, transaction }) {
+    const attrs = span.attributes
+    const baseSegment = transaction.baseSegment
+
+    if (attrs[ATTR_SERVER_ADDRESS]) {
+      baseSegment.addAttribute('host', attrs[ATTR_SERVER_ADDRESS])
```

Comment:
this value also has to be run through the `agent.config.getHostnameSafe` method if it is localhost.  As example from another reconciliation method in this file

```js
        if (urltils.isLocalhost(sanitized)) {
          sanitized = this.agent.config.getHostnameSafe(sanitized)
        }
```